src/test/java/com/o19s/es/ltr/action/BaseIntegrationTest.java

```diff
@@ -98,7 +102,12 @@ public FeatureStoreResponse addElement(StorableElement element) throws Execution
     }
 
     public <E extends StorableElement> E getElement(Class<E> clazz, String type, String name, String store) throws IOException {
-        return new IndexFeatureStore(store, client(), parserFactory()).getAndParse(name, clazz, type);
+        return new IndexFeatureStore(store,
```

Comment:
having a Supplier<Client> here would allow to simply switch to:
```return new IndexFeatureStore(store, this::client, parserFactory()).getAndParse(name, clazz, type);```
instead of pulling a fully-featured QueryShardContext

Reply:
Done!